src/components/Header.tsx

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import { FormEvent, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { addTodo, updateTodo } from '../api/todos';
+
+interface HeaderProps {
+  todos: Todo[],
+  userId: number,
+  handleTempTodo: (todo: Todo | null) => void,
+  handleError: (error: string) => void;
+  handleIsUpdating: (status: boolean) => void,
+  handleUpdatingIds: (ids: number[]) => void,
+}
+
+export const Header = ({
+  todos, userId, handleError, handleTempTodo,
+  handleIsUpdating, handleUpdatingIds,
+}:HeaderProps) => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isDisabled, setIsDisabled] = useState(false);
+  const hasTodos = todos.length > 0;
+
+  const handleCleaner = () => {
+    handleIsUpdating(false);
+    setIsDisabled(false);
+    setTodoTitle('');
+    handleTempTodo(null);
+  };
+
+  const handleUpdateAllCompleted = () => {
+    handleIsUpdating(true);
+    let isAllCompleted = true;
```

Comment:
maybe lets use useState to store and update this value

Reply:
this value is used only in this method so I won't use useState to store it